ingenialink/servo.py

```diff
@@ -41,6 +47,87 @@
 OPERATION_TIME_OUT = -3
 
 
+class DictionaryFactory:
+    """Dictionary factory, creates a dictionary instance depends on the file version
+    and connection interface"""
+
+    _VERSION_ABSOLUTE_PATH = "Header/Version"
+    _VERSION_REGEX = r"(\d+)\.*(\d*)"
+    _MAJOR_VERSION_GROUP = 1
+    _MINOR_VERSION_GROUP = 2
+
+    @classmethod
+    def create_dictionary(cls, dictionary_path: str, interface: Interface) -> Dictionary:
+        """Creates a dictionary instance choosing the class depending on dictionary version and
+         connection interface.
+
+        Args:
+            dictionary_path: target dictionary path
+            interface: connection interface
+
+        Returns:
+            Dictionary instance
+
+        Raises:
+            FileNotFoundError: dictionary path does not exist.
+            ILDictionaryParseError: xdf is not well-formed.
+            ILDictionaryParseError: File is not a xdf.
+            NotImplementedError: Dictionary version is not supported.
+
+        """
+        major_version, minor_version = cls.__get_dictionary_version(dictionary_path)
```

Comment:
The minor version is not used.

Reply:
Could be pushed to the dictionary class

Reply:
I added it because in the future we could need it. The dictionary class has the version as a string.